Lib/pydoc.py

```diff
@@ -2544,9 +2544,42 @@ def bltinlink(name):
             'key = %s' % key, '#ffffff', '#ee77aa', '<br>'.join(results))
         return 'Search Results', contents
 
+    def validate_source_path(path):
+        with warnings.catch_warnings():
+            warnings.filterwarnings('ignore') # ignore problems during import
+            def onerror(modname):
+                pass
+            for importer, modname, ispkg in pkgutil.walk_packages(onerror=onerror):
+                try:
+                    spec = pkgutil._get_spec(importer, modname)
+                except SyntaxError:
+                    # raised by tests for bad coding cookies or BOM
+                    continue
+                loader = spec.loader
+                if hasattr(loader, 'get_source'):
+                    try:
+                        source = loader.get_source(modname)
+                    except Exception:
+                        continue
+                    if hasattr(loader, 'get_filename'):
+                        sourcepath = loader.get_filename(modname)
+                        if path == sourcepath:
+                            return
+                else:
+                    try:
+                        module = importlib._bootstrap._load(spec)
+                    except ImportError:
+                        continue
+                    sourcepath = getattr(module, '__file__', None)
+                    if path == sourcepath:
+                        return
+            else:
+                raise ValueError('not found {found}')
```

Comment:
Hi, I've randomly opened this PR. I'd like to ask you to check this error message, it looks like unfinished.